cmd/odo/odo.go

```diff
@@ -53,7 +54,7 @@ func main() {
 	// Call commands
 	// checking the value of updatenotification in config
 	// before proceeding with fetching the latest version
-	cfg, err := config.New()
+	cfg, err := config.NewGlobalConfig()
```

Comment:
Why is this gloabal config?
Any package outside the pkg/config should in my opinion not refer to either global or local config explicitly in my opinion.
even for commands that request update of a configurable to local or global config explicitly, I would say pass on the states of the respective flags/defaults as applicable onto a single config update function in the pkg/config package and let no outside packages treat them separately? 

Basically external packages shouldn't do in my opinion, NewGlobalConfig or NewLocalConfig may be just NewConfig and pass on everything collected from user down to pkg/config and let all intelligence of config to the pkg/config

Reply:
I can just revert this to `New()` and essentially that proxies to `NewGlobalConfig()` for now ( that will change in the next PR )